compiler/lowerings.nim

```diff
@@ -30,6 +30,23 @@ proc newTupleAccess*(g: ModuleGraph; tup: PNode, i: int): PNode =
     var lit = newNodeIT(nkIntLit, tup.info, getSysType(g, tup.info, tyInt))
     lit.intVal = i
     result[0].add lit
+  elif tup.typ.skipTypes(abstractInst).kind == tyPtr:
+    # iterator items(list: var LetterPairs): lent (Letter, string) =
+    #   for item in list.values:
+    #     yield item
+    # `item` comes from `items(list.values)`  which returns `lent T`,
+    # which implies a `nkHiddenDeref` node when accessing it from yield
+    # so `yield item` is equal to `yield (addr (deref item))`
+    # and `(addr (deref item)` is transformed into `item` 
+    # with a `tyPtr` type in the `transf` phase
+    # and ends up here finally
+    let itemType = tup.typ.skipTypes(abstractInst+{tyPtr})[i]
+    result = newNodeIT(nkHiddenAddr, tup.info, newType(tyPtr, g.idgen, itemType.owner, skipIntLit(itemType, g.idgen)))
+    result.add newNodeIT(nkBracketExpr, tup.info, itemType)
+    result[0].add newTreeIT(nkHiddenDeref, tup.info, tup.typ.skipTypes({tyPtr}), tup)
+    var lit = newNodeIT(nkIntLit, tup.info, getSysType(g, tup.info, tyInt))
+    lit.intVal = i
+    result[0].add lit
```

Comment:
I think this should be fixed in transf.nim instead.

Reply:
=> https://github.com/nim-lang/Nim/pull/23993